# HG changeset patch
# User HackaLittleBit
# Date 1515718483 0
#      Fri Jan 12 00:54:43 2018 +0000
# Branch trunk
# Node ID c689d44e557ed30c22c30330380edaa154799712
# Parent  d8854c7301c7a3d7443f77e969138d97fa5cd762
Fix [FS#6662] Avoid tile operations outside map border.

diff -r d8854c7301c7 -r c689d44e557e src/water_cmd.cpp
--- a/src/water_cmd.cpp	Thu Jan 11 12:04:29 2018 +0000
+++ b/src/water_cmd.cpp	Fri Jan 12 00:54:43 2018 +0000
@@ -73,7 +73,7 @@
  */
 static inline void MarkTileDirtyIfCanalOrRiver(TileIndex tile)
 {
-	if (IsTileType(tile, MP_WATER) && (IsCanal(tile) || IsRiver(tile))) MarkTileDirtyByTile(tile);
+	if (IsValidTile(tile) && IsTileType(tile, MP_WATER) && (IsCanal(tile) || IsRiver(tile))) MarkTileDirtyByTile(tile);
 }
 
 /**
